src/Mod/CppModule.cpp

```diff
@@ -0,0 +1,60 @@
+#define NOMINMAX
+
+#include <filesystem>
+
+#include <Mod/CppModule.hpp>
+
+namespace RC
+{
+    CppModule::CppModule(UE4SSProgram& program, std::wstring&& mod_name, std::wstring&& mod_path) : Mod(program, std::move(mod_name), std::move(mod_path))
+    {
+        m_dlls_path = m_mod_path + L"\\dlls";
+
+        if (!std::filesystem::exists(m_dlls_path))
+        {
+            set_installable(false);
+            return;
+        }
+
+        m_main_dll_module = LoadLibraryW((m_dlls_path + L"\\main.dll").c_str());
+
+        if (!m_main_dll_module)
+        {
+            set_installable(false);
+            return;
+        }
+
+        m_start_mod_func = reinterpret_cast<start_type>(GetProcAddress(m_main_dll_module, "start_mod"));
+        m_uninstall_mod_func = reinterpret_cast<uninstall_type>(GetProcAddress(m_main_dll_module, "uninstall_mod"));
+
+        if (!m_start_mod_func || !m_uninstall_mod_func)
+        {
+            FreeLibrary(m_main_dll_module);
+            m_main_dll_module = NULL;
+
+            set_installable(false);
```

Comment:
We should use `Output::send<LogLevel::Warning>(...)` here to let the user know why the dll couldn't be loaded.
In the future, this message could be sent via `set_installable` instead and outputted alongside the regular `Was unable to install mod` message.